source/buildVersion.py

```diff
@@ -34,7 +34,12 @@ def _updateVersionFromVCS():
 
 
 def formatDevVersionString():
-	return "%s.%s.%sdev"%(version_year,version_major,version_minor)
+	return "{y}.{M}.{m}dev".format(y=version_year, M=version_major, m=version_minor)
+
+def formatVersionString(withBuildPart = False):
+	if withBuildPart:
+		return "{y}.{M}.{m}.{b}".format(y=version_year, M=version_major, m=version_minor, b=version_build)
+	return "{y}.{M}.{m}".format(y=version_year, M=version_major, m=version_minor)

```

Comment:
> I think it's a good idea to keep the concepts of api version and nvda separate.

This makes sense, as long as we explicitly state "NVDA api version" (not NVDA version) when mentioning the version number in the incompatible dialogs.

Reply:
I've been thinking about this over the last few days, I'm questioning this a little. `addonAPIVersion.CURRENT` will always equal the version number, `addonAPIVersion.BACK_COMPAT_TO` will always be equal to a some past `addonAPIVersion.CURRENT` version. For end users it's probably easiest for these to be referred to as NVDA versions, otherwise we risk confusion along the lines of "well what version of NVDA do I have to install to get THAT API version", but most likely they will assume they are equal anyway. For add-on authors, its important to realise that its the API changes that they need to care about, rather than the NVDA versions, this is a subtle difference. I'll look at implementing changing this so that one is implemented in terms of the other. There is also the distinction of end user (E.G. in the GUI) facing version strings, and developer facing ones (exe properties, log files)